example/.mage/tools.mk

```diff
@@ -1,49 +1,11 @@
 mage_folder := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
-mage_generated_path := $(mage_folder)/tools/mgmake
 mage_tools_path := $(mage_folder)/tools
-mage_targets_file := $(mage_generated_path)/targets.mk
-mage := $(mage_generated_path)/local-mage
-mgmake := $(mage_folder)/mgmake_gen.go
-mgmain := $(mage_folder)/mgmain_gen.go
+mage := $(mage_tools_path)/mgmake/magefile
+version := $(shell cd $(mage_folder) && go list -m -f '{{ .Version }}' go.einride.tech/mage-tools)
 
-define mgmake_content
-// Code generated by Mage-tools. DO NOT EDIT.
-//go:build mage
-// +build mage
-
-package main
-
-// mage:import
-import _ "go.einride.tech/mage-tools/mgmake"
-endef
-
-define mgmain_content
-// Code generated by Mage-tools. DO NOT EDIT.
-//go:build ignore
-// +build ignore
-
-package main
-
-import (
-	"os"
-	"github.com/magefile/mage/mage"
-)
-
-func main() { os.Exit(mage.Main()) }
-endef
-
-include $(mage_targets_file)
-
-$(mage_targets_file): $(mage_folder)/go.mod $(mage_folder)/*.go
-	@git clean -fdx $(mage_generated_path)
-	@mkdir -p $(mage_generated_path)
-	$(file > $(mgmake),$(mgmake_content))
-	$(file > $(mgmain),$(mgmain_content))
-	@cd $(mage_folder) && \
-		go mod tidy && \
-		go run $(notdir $(mgmain)) -compile $(mage) && \
-		$(mage) generateMakefile $(@)
+$(mage): $(mage_folder)/go.mod $(mage_folder)/*.go
+	@go run go.einride.tech/mage-tools@$(version) gen
```

Comment:
the `@version` should not be needed if this is run from the `.mage` directory, the version should be picked up by `go.mod`

Reply:
Fixed